manual/src/cmds/runtime-tracing.etex

```diff
@@ -314,4 +319,144 @@ Percentile 	 Latency (ms)
 99.9990 	 9.17
 99.9999 	 9.17
 100.0000 	 9.17
-\end{verbatim}
\ No newline at end of file
+\end{verbatim}
+
+\section{s:runtime-tracing-custom-events}{Custom events}
+
+\subsection{s:runtime-tracing-custom-events-overview}{Overview}
+
+The runtime events system supports defining custom events. Identified by a
+globally unique name, they are emitted with payloads of built-in (\texttt{unit},
+\texttt{int}, \texttt{span}) and user-defined types. To understand the
+manipulation of custom events, it is useful to know how they are transported and
+stored: their representation consists in a name string (in fact, an index into
+an array of all custom names) and an arbitrary array of bytes. Custom event
+types can be \emph{registered} by providing encoding and decoding functions to
+and from an array of bytes, via the function \texttt{Type.register}.
+
+Defining a new custom event (whether its payload is of a built-in type or a
+custom type) is done via another registration function,
+\texttt{Runtime_events.User.register}. This function records the association
+between the custom event's name, its type, and a \emph{tag}. The tag is then
+used when emitting or consuming custom events; it acts as a shorthand for the
+event name.
+
+Once an event is registered, values for that event are emitted using
+\texttt{Runtime_events.User.write}.
+
+To summarize, to emit and consume custom events of a custom type (for example),
+the user needs to:
+
+\begin{enumerate}
+\item register the custom type
+\item extend the \texttt{User.tag} variant with a new constructor
+\item register the new custom event, binding together the custom event name, the
+  new tag, and the new custom type
+\item emit instances of the event using \texttt{Runtime_events.User.write}
+\item create an event cursor (or use an existing one) and use
+  \texttt{Callbacks.add_user_event} to indicate a callback which should receive
+  all events of the custom type.
+\item \emph{(Optional, if there are distinct events registered with the same
+  custom type:)} In that callback, pattern-match on the event tag in order to
+  discriminate between different event tags of the same custom type.
+\end{enumerate}
+
+Note that if the emitter and the consumer are different programs, both must
+perform steps 1 to 3 to register custom events and custom event types (if any).
+
+Unregistered events for user-defined types are currently ignored. As a fallback,
+unregistered, custom events of a built-in type are available, but are all
+attributed the \texttt{UNK} (unknown) tag.
```

Comment:
```suggestion
given the \texttt{UNK} (unknown) tag.
```
or
```suggestion
tagged \texttt{UNK} (unknown).
```

Reply:
I integrated your second suggestion in 9d1768b.